pynitrokey/cli/nk3/piv.py

```diff
@@ -14,725 +14,734 @@
+    @click.option(
+        "--path",
+        type=click.Path(allow_dash=True),
+        default="-",
+        help="Read certificate from path.",
     )
+    def read_certificate(format: str, key: str, path: str) -> None:
+        device = PivApp()
 
-    device.send_receive(0xDB, 0x3F, 0xFF, payload)
-
-
-@piv.command(help="Read a certificate from a key slot.")
-@click.option(
-    "--format",
-    type=click.Choice(["DER", "PEM"], case_sensitive=False),
-    default="PEM",
-    help="Format of certificate.",
-)
-@click.option(
-    "--key",
-    type=click.Choice(
-        [
-            "9A",
-            "9C",
-            "9D",
-            "9E",
-            "82",
-            "83",
-            "84",
-            "85",
-            "86",
-            "87",
-            "88",
-            "89",
-            "8A",
-            "8B",
-            "8C",
-            "8D",
-            "8E",
-            "8F",
-            "90",
-            "91",
-            "92",
-            "93",
-            "94",
-            "95",
-        ],
-        case_sensitive=False,
-    ),
-    default="9A",
-    help="Key slot for operation.",
-)
-@click.option(
-    "--path",
-    type=click.Path(allow_dash=True),
-    default="-",
-    help="Read certificate from path.",
-)
-def read_certificate(format: str, key: str, path: str) -> None:
-    device = PivApp()
-
-    value = device.cert(bytes(bytearray.fromhex(KEY_TO_CERT_OBJ_ID_MAP[key.upper()])))
-
-    if value is None:
-        print("Certificate not found", file=sys.stderr)
-        return
-
-    format = format.upper()
-    if format == "DER":
-        cert_serialized = value
-        cryptography.x509.load_der_x509_certificate(value)
-    elif format == "PEM":
-        cert = cryptography.x509.load_der_x509_certificate(value)
-        cert_serialized = cert.public_bytes(Encoding.PEM)
-
-    with click.open_file(path, mode="wb") as f:
-        f.write(cert_serialized)
+        value = device.cert(
+            bytes(bytearray.fromhex(KEY_TO_CERT_OBJ_ID_MAP[key.upper()]))
+        )
+
+        if value is None:
+            print("Certificate not found", file=sys.stderr)
+            return
+
+        format = format.upper()
+        if format == "DER":
+            cert_serialized = value
+            cryptography.x509.load_der_x509_certificate(value)
+        elif format == "PEM":
+            cert = cryptography.x509.load_der_x509_certificate(value)
+            cert_serialized = cert.public_bytes(Encoding.PEM)
+
+        with click.open_file(path, mode="wb") as f:
+            f.write(cert_serialized)
+
+except ImportError:
+    from pynitrokey.cli.nk3.pcsc_absent import PCSC_ABSENT
+
+    @nk3.group(
+        invoke_without_command=True,
+        context_settings=dict(
+            ignore_unknown_options=True,
+        ),
+    )
+    @click.argument("args", nargs=-1, type=click.UNPROCESSED)
+    def piv(args: list[str]) -> None:
+        """Nitrokey PIV App"""
+        local_critical(PCSC_ABSENT)
```

Comment:
I would set `support_hint=False` because this is not an unexpected error (so we don’t need the logs) and it is easy to miss the actual error message in all that noise.